snap/snapcraft.yaml

```diff
@@ -149,28 +161,27 @@ parts:
     # so then 'pip install PyYAML' gets cross.
     python-packages: [wheel]
     build-environment:
-      - LDFLAGS: -L/usr/lib/python3.8
-      - CPPFLAGS: -I/usr/include/python3.8
+      - LDFLAGS: -L/usr/lib/python3.10
+      - CPPFLAGS: -I/usr/include/python3.10
     override-pull: |
-      # do the usual pull stuff
-      snapcraftctl pull
+      ${SNAP}/libexec/snapcraft/craftctl default
```

Comment:
it's odd that you'd have to spell out the full path to `craftctl` like this, how come?

Reply:
oh, interesting, all crafts might suffer from this, it might be to avoid using the craftctl from this build itself

Reply:
In the `PRIME` step, the `PATH` changed to `PATH=/root/parts/python3/install/usr/bin:/root/stage/usr/bin:/root/stage/bin:/snap/snapcraft/x1/libexec/snapcraft:/snap/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin`, where there is another `craftctl` located at `/root/stage/bin/craftctl`. And that one doesn't work because `craft-parts` not exist.

Reply:
Maybe you need to `organize` the `craftctl` into `libexec`, like you did in rockcraft?
```
    organize:
        bin/craftctl: libexec/rockcraft/craftctl

Reply:
> Maybe you need to `organize` the `craftctl` into `libexec`, like you did in rockcraft?
> 
> ```
>     organize:
>         bin/craftctl: libexec/rockcraft/craftctl
> ```

Yeah, it work without need the full path, since it is not in the path.

Reply:
But the actually issue is `craftctl default` result is different than the non-override result.

The linking error not happened if `override-prime` not exist. But if you put:
```
    override-prime: |
      craftctl default
```
The `enable-patchelf` not work for some binaries.

Reply:
Yes I can confirm that if you don't override-prime, the "patchelfing" works.
From what I can tell, the patchelf feature in Snapcraft is implemented as a prime callback, so maybe there's a bug in the interaction between callbacks and `craftctl`
ping @cmatsuoka

Reply:
@syu-w I think the cleanest course of action here is to actually tear most of this `snapcraft.yaml` down and rewrite it in a way that's similar to Rockcraft and Snapcraft, like Sergio suggested. I have the feeling that there's a lot of cruft here that is no longer necessary and is making our understanding of what's going on impossible

Reply:
yes, @syu-w , please as stated 3 days ago, copy what is in rockcraft and snapcraft's snapcraft.yaml, even the part names, and add the special things needed for charmcraft in a separate commit. There are things that do not make sense here.